### PR TITLE
[ContactStructuralMechanicsApplication] Typo, model called main_model_part

### PR DESCRIPTION
**Pull request by loumalouomega**
I was using the model, but called main_model_part

### CODE REVIEW

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py [hunk 1]
```diff
@@ -187,18 +193,20 @@ def print_warning_on_rank_zero(self, *args):
 
     def _create_linear_solver(self):
         linear_solver = super(ContactImplicitMechanicalSolver, self)._create_linear_solver()
+        if (self.contact_settings["rescale_linear_solver"].GetBool() is True):
+            linear_solver = KM.ScalingSolver(linear_solver, False)
         mortar_type = self.contact_settings["mortar_type"].GetString()
         if (mortar_type == "ALMContactFrictional" or mortar_type == "ALMContactFrictionlessComponents"):
             if (self.contact_settings["use_mixed_ulm_solver"].GetBool() == True):
-                self.print_on_rank_zero("::[Contact Mechanical Implicit Dynamic Solver]:: ", "Using MixedULMLinearSolver, definition of ALM parameters recommended")
+                self.print_on_rank_zero("::[Contact Mechanical Static Solver]:: ", "Using MixedULMLinearSolver, definition of ALM parameters recommended")
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py]: ?

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py [hunk 2]
```diff
@@ -213,12 +221,12 @@ def _create_linear_solver(self):
...
                         amgcl_param["block_size"].SetInt(self.main_model_part.ProcessInfo[KM.DOMAIN_SIZE])
-                        amgcl_param = self.settings["linear_solver_settings"].RecursivelyValidateAndAssignDefaults(amgcl_param)
-                        linear_solver = KM.AMGCLSolver(amgcl_param)
+                        self.linear_solver_settings.RecursivelyValidateAndAssignDefaults(amgcl_param)
+                        linear_solver = KM.AMGCLSolver(self.linear_solver_settings)
                     mixed_ulm_solver = CSMA.MixedULMLinearSolver(linear_solver, self.contact_settings["mixed_ulm_solver_parameters"])
                     return mixed_ulm_solver
                 else:
-                    self.print_on_rank_zero("::[Contact Mechanical Implicit Dynamic Solver]:: ", "Mixed solver not available: "+name_mixed_solver+". Using not mixed linear solver")
+                    self.print_on_rank_zero("::[Contact Mechanical Static Solver]:: ", "Mixed solver not available: " + name_mixed_solver + ". Using not mixed linear solver")
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py]: ?

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py [hunk 3]
```diff
@@ -46,6 +44,7 @@ def __init__(self, main_model_part, custom_settings):
                 "adaptative_strategy"                               : false,
                 "split_factor"                                      : 10.0,
                 "max_number_splits"                                 : 3,
+                "inner_loop_iterations"                             : 5,
```
Comment (philbucher) [inline applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py:47]: this has nothing to do with the title of the PR

**Review by philbucher — COMMENTED** [review]
There are some unrelated changes. Fine for me but approval usually takes longer => I do separate PRs to make the review process easier

also why do you rename to `Static` if it is the dynamic?

**Comment by loumalouomega** [comment]
I was working in a different branch, those changes will come with the, #2808

**Comment by philbucher** [comment]
should I approve them?

File: applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py (hunk 1 above)
Comment (loumalouomega) [reply applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_implicit_dynamic_solver.py]: Copy paste error

**Comment by loumalouomega** [comment]
Yes please, I will correct the typos in the other PR

**Review by philbucher — DISMISSED** [review]
so you want to commit the typos just to fix them later i a different PR? *confused*

**Comment by loumalouomega** [comment]
Solved

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**